src/software/thunderscope/gl/layers/gl_movement_field_test_layer.py

```diff
@@ -0,0 +1,104 @@
+import math
+from pyqtgraph.Qt.QtCore import Qt
+from software.thunderscope.gl.helpers.extended_gl_view_widget import MouseInSceneEvent
+from proto.import_all_protos import *
+from software.thunderscope.gl.layers.gl_world_layer import tbots_cpp
+from software.thunderscope.proto_unix_io import ProtoUnixIO
+from software.logger.logger import create_logger
+from software.thunderscope.gl.layers.gl_layer import GLLayer
+from software.thunderscope.thread_safe_buffer import ThreadSafeBuffer
+
+logger = create_logger(__name__)
+
+
+class GLMovementFieldTestLayer(GLLayer):
+    def __init__(
+        self, name: str, fullsystem_io: ProtoUnixIO, buffer_size: int = 5
+    ) -> None:
+        """Initialize the GLMovementFieldTestLayer
+
+        :param name:            The displayed name of the layer
+        :param buffer_size:     The buffer size we have
+        :param fullsystem_io:   The fullsystem protounix io
+        """
+        super().__init__(name)
+
+        self.world_buffer: ThreadSafeBuffer = ThreadSafeBuffer(buffer_size, World)
+        self.fullsystem_io: ProtoUnixIO = fullsystem_io
+        self.selected_robot_id = 0
+        self.cached_team: tbots_cpp.Team = None
+        self.is_selected = False
+
+    def select_closest_robot(self, point):
+        """Find the closest robot to a point
+
+        :param point: the reference point to find the closest robot
+        """
+        closest_robot = self.cached_team.getNearestRobot(
+            tbots_cpp.Point(point.x(), point.y())
+        )
+        if closest_robot is None:
+            logger.warning(
+                "No robot founds. Are you sure there is friendly robot on field?"
+            )
```

Comment:
to prevent unintended access of `closest_robot`
```suggestion
            logger.warning(
                "No robot founds. Are you sure there is friendly robot on field?"
            )
            return
```